.github/workflows/dead-link-checker.yaml

```diff
@@ -0,0 +1,36 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: Dead Link Checker
+
+on:
+  pull_request:
+
+concurrency:
+  group: dlc-${{ github.event.pull_request.number || github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  CheckDeadLinks:
+    runs-on: ubuntu-latest
+    timeout-minutes: 30
```

Comment:
We only need to run this if select dirs have changes (e.g. `site3/`) and can skip for majority of the PRs

Reply:
Hi, I've thought of doing that, but if someone adds MD files to another dir, that may result in frequent maintenance of this directory (most times. people probably forget to do this).

It's a simple CI, so I'd rather run it for every PR, WDYT?

Reply:
now,the check is only triggered when the MD file is modified

Reply:
I didn't notice before that there is such a way to determine the file type 🤣